Source/WebKit/UIProcess/XR/PlatformXRSystem.cpp

```diff
@@ -47,28 +47,38 @@ WTF_MAKE_TZONE_ALLOCATED_IMPL(PlatformXRSystem);
 PlatformXRSystem::PlatformXRSystem(WebPageProxy& page)
     : m_page(page)
 {
-    m_page->protectedLegacyMainFrameProcess()->addMessageReceiver(Messages::PlatformXRSystem::messageReceiverName(), m_page->webPageIDInMainFrameProcess(), *this);
```

Comment:
The MESSAGE_CHECK above uses `m_page` without a null-check. How can we fix that? Only do the message check if m_page is alive?

Reply:
WebPageProxy ultimately owns PlatformXRSystem. So we "know" that m_page is alive. The old code demonstrated that using WeakRef.

You could stick with WeakRef.

The idiomatic way to make this clear is to have PlatformXRSystem forward ref()/deref() to m_page, and have WebPageProxy point to PlatformXRSystem using a unique_ptr.

Reply:
We cannot forward ref-counting because of this:
```
internals().xrSystem = nullptr;
```

As a result, PlatformXRSystem needs its own ref-counting and may indeed outlive its page,.

Reply:
With regards to the message checks, this is easily fixable. We only need the page to get the IPC connection. However, the generated code can pass the IPC::Connection to us, we don't need to retrieve it from the page.

For each function that uses MESSAGE_CHECK, add a first parameter: `IPC::Connection& connection, `. The generated bindings code will know to pass in the IPC connection if this parameter exists.

Then update the macros above so that the callers pass in the connection object.

Reply:
Got it, I just updated the code as recommended. I'm waiting on the bots to see if it builds since this code doesn't get built locally on macOS.